### PR TITLE
Consolidate zen_get_uprid

### PR DESCRIPTION
**Pull request by mc12345678**
Now that the admin and catalog finally provide the same
uprid on the admin as it does on the catalog and there is
a consolidation available, this is the simplified (refactored)
version of the function used on both catalog and admin side.

### CODE REVIEW

**Comment by drbyte** [comment]
I notice the generated output no longer includes `(` and `)` parentheses on the admin side. Thoughts?

File: includes/functions/functions_general_shared.php [hunk 1]
```diff
@@ -81,6 +81,31 @@ function zen_round($value, $precision) {
...
+/**
+ * Return a product ID with attributes hash
+ * @param string|int $prid
+ * @param array|string $params
+ * @return string
+ */
+  function zen_get_uprid($prid, $params) {
+    $uprid = $prid;
+    if (!is_array($params) || strstr($prid, ':')) return $prid;
+
```
Comment (scottcwilson) [inline includes/functions/functions_general_shared.php]: $lb = ''; $rb = ''; 
if (IS_ADMIN_FLAG) { 
   $lb = '('; 
   $rb = ')'; 
}

**Comment by scottcwilson** [comment]
Could just do a check for  IS_ADMIN_FLAG create left_bracket and right bracket characters (empty string on catalog side, brackets on admin side).   Then use them in setting $uprid.

**Comment by scottcwilson** [comment]
Still a consolidation win!  Thanks for doing this @mc12345678.

**Comment by drbyte** [comment]
Hmmm ... looks like the function is never used admin-side anyway.

**Comment by lat9** [comment]
> 
> Hmmm ... looks like the function is never used admin-side anyway.

I was just going to post that.  I'll note, though, that I've got a couple of plugins that _do_ use that function on the admin side.

**Comment by drbyte** [comment]
> I was just going to post that. I'll note, though, that I've got a couple of plugins that _do_ use that function on the admin side.

👍 
Yes, that's why I decided to merge, since this makes it still available in both places.

**Comment by mc12345678** [comment]
I would have used the function years ago for an application, but because some other existing plugins happily used it causing elimination of the hash portion of the record I couldn't tell if the uprid was on its way in and not fully incorporated or on its way out and being phased. 

As far as the single quotes that were around the content in the admin, I have yet to see how it had any functional effect though it seemed to make it easier to read when following changes from ZC version to version.

**Pull request merged**